### PR TITLE
Validate that the child key in a relationship is not a primary key

### LINKED ISSUE #1236: Validate that relationship must be between a `primary key` and `foreign key`
Version: SDV 1.0 (In development)

### Problem Description
Currently, the metadata allows you add a relationship between two primary keys, as shown in #1232. This should not be allowed.

If you have a column that refers to another table, it _must_ be a foreign key, not primary key.

### Expected behavior
Running the code from #1232, we should throw an Error if you are specifying a relationship between 2 primary keys.

```python
table = pd.DataFrame({
    'pk': list(range(100)),
    'col1': np.random.random(100),
    'col2': np.random.choice(['a', 'b', 'c'], 100)
})
data = {'table': table, 'table2': table}

metadata = MultiTableMetadata()
metadata.detect_table_from_dataframe('table', table)
metadata.set_primary_key('table', 'pk')

metadata.detect_table_from_dataframe('table2', table)
metadata.set_primary_key('table2', 'pk')

metadata.add_relationship('table', 'table2', 'pk', 'pk')
```
Output:
```
Invalid Metadata Error: A relationship must be specified between a primary key and non-primary key.
```

### Additional context
The full specification is: A relationship _must_ be between a primary key and a non-primary key.

We should also validate this this in the `MultiTableMetadata.validate` method.

### PR DESCRIPTION
**Pull request by fealho**
Resolve #1236.

### CODE REVIEW

**Comment by npatki** [comment]
@fealho Just saw the title of this PR. The child's key actually _must_ be a foreign key.

In a parent-child relationship:

1. **Parent Table** has the primary key
2. **Child Table** has the foreign key

**Comment by fealho** [comment]
@npatki Thanks! My brain lagged, I meant to say `is not a primary key`.

File: tests/unit/metadata/test_multi_table.py [hunk 1]
```diff
@@ -818,6 +837,37 @@ def test_validate_raises_errors(self):
         with pytest.raises(InvalidMetadataError, match=error_msg):
             instance.validate()
 
+    def test_validate_child_key_is_primary_key(self):
+        """Test it crashes if the child key is a primary key."""
+        table = pd.DataFrame({
```
Comment (pvk-developer) [inline tests/unit/metadata/test_multi_table.py:842]: missing `# Setup`

**Review by pvk-developer — APPROVED** [review]
Just a minor nit, but other than that looks good

**Review by amontanez24 — APPROVED** [review]
LGTM!

**Pull request merged**